nautobot/extras/models/jobs.py

```diff
@@ -290,10 +295,13 @@ def job_task(self):
 
     def clean(self):
         """For any non-overridden fields, make sure they get reset to the actual underlying class value if known."""
-        if self.job_class is not None:
+        from nautobot.extras.jobs import get_job
+
+        job_class = get_job(self.class_path, reload=True)
```

Comment:
In the UI edit case this will result in reloading the jobs twice - once in the form clean(), once in the model clean(). Might be possible to refactor the form clean so that it doesn't need to do this?